src/ngRoute/route.js

```diff
@@ -786,7 +788,10 @@ function $RouteProvider() {
         if (angular.isDefined(template)) {
           locals['$template'] = template;
         }
-        return $q.all(locals);
+        return $q.all(locals).then(function(locals) {
+          $browser.$$completeOutstandingRequest(function() {});
```

Comment:
I would use `angular.noop` here. Actually, I would initialize `noop = angular.noop` (e.g. the same as `isArray`) above.